python/pyarrow/_dataset.pyx

```diff
@@ -4111,7 +4113,9 @@ class ScanNodeOptions(_ScanNodeOptions):
     **kwargs : dict, optional
         Scan options. See `Scanner.from_dataset` for possible arguments.        
     require_sequenced_output : bool, default False
-        Assert implicit ordering on data.
+        Batches are yielded sequentially, like single-threaded
```

Comment:
Is this still needed?

Reply:
Yes, the old statement is wrong.

Reply:
Sorry for not making myself clear enough. By `still needed` I mean the existence of the parameter `require_sequenced_output`. It seems that when introduced in #44083 , it is determined by the parameter `require_sequenced_output` of `_dataset_to_decl`. Since now you have replaced the `require_sequenced_output` with `implicit_ordering` in `_dataset_to_decl`, I wonder if this could be removed as well. Or is "we should not remove a parameter of a public API" the consideration here?

Thanks.

Reply:
I see. The `require_sequenced_output` of `ScanNodeOptions` has been existing in C++ for a few years, but it was missing from Python API. Makes sense to expose it to Python. It might be useful for others. So I would keep it in.

Reply:
That makes sense. Thank you.